libr/core/cmd_write.c

```diff
@@ -2025,8 +2025,29 @@ static int cmd_write(void *data, const char *input) {
 		wA_handler_old (core, input + 1);
 		break;
 	case ' ': // "w"
-		w_handler_old (core, input + 1);
+	{
+		size_t len = core->blocksize;
+		const char *current_charset = r_config_get (core->config, "cfg.charset");
+		if (R_STR_ISEMPTY (current_charset)) {
+			w_handler_old (core, input + 1);
+		} else {
+			if (len > 0) {
+				size_t in_len = strlen (input + 1);
+				ut8 *out = malloc (len); //suppose in len = out len TODO: change it
```

Comment:
imho this logic shouldbe implemented inside w_handler_old() just when trying to write the string